src/packages/src/xroad/redhat/SPECS/xroad-secret-store-local.spec

```diff
@@ -0,0 +1,148 @@
+%include %{_specdir}/common.inc
+# produce .elX dist tag on both centos and redhat
+%define dist %(/usr/lib/rpm/redhat/dist.sh)
+
+Name:               xroad-secret-store-local
+Version:            %{xroad_version}
+# release tag, e.g. 0.201508070816.el7 for snapshots and 1.el7 (for final releases)
+Release:            %{rel}%{?snapshot}%{?dist}
+Summary:            Meta-package for X-Road remote secret store dependencies
+Group:              Applications/Internet
+License:            MIT
+Requires:           jq, bao >= 2.0.0
+Requires:           xroad-base = %version-%release
+Conflicts:          xroad-secret-store-local-remote
+
+%description
+X-Road OpenBao Auto Unseal Service
+
+%clean
+rm -rf %{buildroot}
+
+%prep
+
+%build
+
+%install
+mkdir -p %{buildroot}%{_unitdir}
+mkdir -p %{buildroot}/usr/share/xroad/scripts/
+mkdir -p %{buildroot}/etc/xroad/services/
+
+cp -p %{_sourcedir}/secret-store-local/xroad-secret-store-local.service %{buildroot}%{_unitdir}
+cp -p %{srcdir}/common/xroad-secret-store-local/etc/xroad/services/secret-store-local.conf %{buildroot}/etc/xroad/services/
+cp -p %{srcdir}/common/xroad-secret-store-local/usr/share/xroad/scripts/secret-store-unseal.sh %{buildroot}/usr/share/xroad/scripts/
+cp -p %{srcdir}/common/xroad-secret-store-local/usr/share/xroad/scripts/secret-store-setup.sh %{buildroot}/usr/share/xroad/scripts/
+
+%files
+%defattr(0640,xroad,xroad,0751)
+%attr(644,root,root) %{_unitdir}/xroad-secret-store-local.service
+%config /etc/xroad/services/secret-store-local.conf
+%attr(554,root,xroad)  /usr/share/xroad/scripts/secret-store-unseal.sh
+%attr(554,root,xroad) /usr/share/xroad/scripts/secret-store-setup.sh
+
+%pre -p /bin/bash
+%upgrade_check
+
+set -e
+
+# Function to handle errors - only clean up on failure
+cleanup() {
+    if [ $? -ne 0 ]; then
+        echo "Installation failed, cleaning up..."
+        if [ -d "/opt/openbao/tls" ]; then
+            rm -f /opt/openbao/tls/tls.{key,crt} 2>/dev/null || true
+        fi
+        rm -f /etc/pki/ca-trust/source/anchors/openbao.crt 2>/dev/null || true
+    fi
+}
+
+trap cleanup EXIT
+
+if [ $1 -eq 1 ] || [ $1 -eq 2 ]; then  # 1 = fresh install, 2 = upgrade
```

Comment:
Can we extract duplicate initialization code and share it between deb/rpm? This would also go inline with the idea of adding init scripts later